### PR TITLE
Adding Milliseconds to DateTime fields

### PR DESCRIPTION
**Pull request by benrr101**
This is a slightly larger change than anticipated due the difference between `DATETIME`, `DATETIME2`, and `DateTime`. The `DATETIME` type always uses 3 decimal points of a second, while the `DATETIME2` type has 7 (although since `DATETIME2(7)` is default in SSMS suggesting that it is a variable precision type). Regardless of the db type, the engine returns `DateTime` C# type. The db types are only made visible via the column info, via the numeric precision and numeric scale. My findings were as such:
`DATETIME `: Precision = 23, Scale = 3
`DATETIME2`: Precision = 255, Scale = 7

The scale corresponds neatly with the number of second decimal points to show. The buffer file writer was modified to store both the scale and the number of ticks. Then the buffer file reader was modified to read in the precision and the number of ticks and generate the ToString version of the DateTime to add "f" as many times as there is scale, which corresponds to milliseconds.

### CODE REVIEW

**Comment by benrr101** [comment]
Anyone who wanted to see what it looks like now
millisecondview

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/ServiceBufferFileStreamReader.cs [hunk 1]
```diff
@@ -260,11 +260,21 @@ public FileStreamReadResult ReadDecimal(long offset)
...
+            return ReadCellHelper(offset,
+                length =>
+                {
+                    precision = BitConverter.ToInt32(buffer, 0);
+                    long ticks = BitConverter.ToInt64(buffer, 4);
+                    return new DateTime(ticks);
+                }, null,
+                time =>
+                {
+                    // Output the number milliseconds equivalent to the precision
```
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/ServiceBufferFileStreamReader.cs]: For the dumb among us, mention that new string('f', precision) output ffff for precision = 4

**Comment by kevcunnane** [comment]
:shipit:

**Comment by benrr101** [comment]
For proof that the DateTime2(0) works
vscodedatetime2 0

**Pull request merged**